examples/cpp/helloworld/other_service.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#ifndef OTHER_SERVICE_H 
+#define OTHER_SERVICE_H
+
+// See https://github.com/protocolbuffers/protobuf/blob/master/cmake/README.md#dlls-vs-static-linking
+#define PROTOBUF_USE_DLLS
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <grpcpp/ext/proto_server_reflection_plugin.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/health_check_service_interface.h>
+
+#ifdef BAZEL_BUILD
+#include "examples/protos/helloworld.grpc.pb.h"
+#else
+#include "helloworld.grpc.pb.h"
+#endif
+
+#include "IDynamicService.h"
+
+using grpc::Server;
+using grpc::ServerBuilder;
+using grpc::ServerContext;
+using grpc::Status;
+using helloworld::Greeter;
+using helloworld::HelloReply;
+using helloworld::HelloRequest;
+
+// Logic and data behind the server's behavior.
+class OtherServiceImpl final : public Greeter::Service { //, public IDynamicService {
+ public:
+  Status SayHello(
+    ServerContext* context,
+    const HelloRequest* request,
+    HelloReply* reply
+  );
+};
+
+extern "C" {
+  __declspec(dllexport) IDynamicService* CreateOtherServiceHelper();
```

Comment:
I created these two different mechanisms: one where the memory is owned by the DLL and returned, and another where the memory is located in the server and assigned in the DLL. Trying to rule out memory bounds errors here